File: src/OSPSuite.SimModel/src/SolverWarning.cpp
#ifdef _WINDOWS_PRODUCTION
#pragma managed(push,off)
#endif

#include "SimModel/SolverWarning.h"

#ifdef _WINDOWS_PRODUCTION
#pragma managed(pop)
#endif

namespace SimModelNative
{

using namespace std;

SolverWarning::SolverWarning(double time, string message)
{
	_time = time;
	_message = message;
}

SolverWarning::SolverWarning(string message)
{
	_time = 0.0;
	_message = message;
}

double SolverWarning::Time() const
{
	return _time;
}

std::string SolverWarning::Message() const
{
	return _message;
}


}//.. end "namespace SimModelNative"

File: src/OSPSuite.SimModel/SolverWarning.cs
namespace OSPSuite.SimModel
{
   public class SolverWarning
   {
      internal SolverWarning(double outputTime,string warning)
      {
         OutputTime = outputTime;
         Warning = warning;
      }

      public double OutputTime { get; }
      public string Warning { get; }
   }
}
